test/resources/knativeresources.go

```diff
@@ -81,3 +88,38 @@ func IsKnativeDeploymentReady(dpList *v1.DeploymentList, expectedDeployments []s
 	}
 	return true, nil
 }
+
+// GetExpectedDeployments will return an array of deployment resources based on the version for the knative
+// component.
+func GetExpectedDeployments(t *testing.T, version, kcomponent string) []string {
+	manifestPath := common.RetrieveManifestPath(version, kcomponent)
+	manifest, err := mf.NewManifest(manifestPath)
+	if err != nil {
+		t.Fatalf("Failed to get the manifest for Knative: %v", err)
+	}
+
+	deployments := []string{}
+	for _, resource := range manifest.Filter(mf.ByKind("Deployment")).Resources() {
+		deployments = append(deployments, resource.GetName())
+	}
+	return removeDuplications(deployments)
+}
+
+// SetKodataDir will set the env var KO_DATA_PATH into the path of the kodata of this repository.
+func SetKodataDir() {
+	_, b, _, _ := runtime.Caller(0)
+	koPath := filepath.Join(filepath.Dir(b)+"/../..", "cmd/operator/kodata")
```

Comment:
../.. is a bit of a magic string, is there a way we can get the top level dir without needing to know how nested we are?